client/components/domains/register-domain-step/index.jsx

```diff
@@ -385,6 +385,29 @@ const RegisterDomainStep = React.createClass( {
 					return suggestion.domain_name;
 				} );
 
+				if ( abtest( 'domainSuggestionNudgeLabels' ) === 'withLabels' ) {
+					const isFree = ( suggestion ) => ( suggestion.is_free === true ),
+						exactMatchBeforeTld = ( suggestion ) => (
+							startsWith( suggestion.domain_name, `${ domain.replace( / /g, '' ) }.` ) &&
```

Comment:
Ugh, for my suggestion about only using the exact match before TLD to work, you also need to strip the domain of the TLD part. Otherwise, you'll never get the `Recommended` flag to show up for direct domain searches.